### PR TITLE
Remove some Python 2 compatibility code.

### PR DESCRIPTION
**Pull request by robertnishihara**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -143,13 +143,13 @@ script:
...
-  - if [ $RAY_CI_PYTHON_AFFECTED == "1" ]; then python -c 'import sys;exit(sys.version_info>=(3,5))' || RAY_FORCE_DIRECT=0 python -m pytest -v --durations=5 --timeout=300 python/ray/tests/py3_test.py; fi
+  - if [ $RAY_CI_PYTHON_AFFECTED == "1" ]; then RAY_FORCE_DIRECT=0 python -m pytest -v --durations=5 --timeout=300 python/ray/experimental/test/async_test.py; fi
+  - if [ $RAY_CI_PYTHON_AFFECTED == "1" ]; then RAY_FORCE_DIRECT=0 python -m pytest -v --durations=5 --timeout=300 python/ray/tests/py3_test.py; fi
 
   # py bazel tests, run using local strategy since PY2 breaks with sandbox
   - ./ci/keep_alive bazel test --spawn_strategy=local --flaky_test_attempts=3 --python_version=$BAZEL_PYTHON_VERSION --nocache_test_results --test_verbose_timeout_warnings --incompatible_allow_python_version_transitions=false --incompatible_py3_is_default=false --progress_report_interval=100 --show_progress_rate_limit=100 --show_timestamps --test_output=errors --test_tag_filters=-jenkins_only python/ray/...
```
Comment (ericl) [inline .travis.yml]: can remove --spawn_strategy=local, all the python flags, and also the code above that sets BAZEL_PYTHON_VERSION

**Review by pcmoritz — APPROVED** [review]
Looks good, you should also make the changes Eric suggested.

**Comment by robertnishihara** [comment]
`test_debug_tools` has failed a couple of times in Linux, so need to look into that before merging.

**Comment by ericl** [comment]
It's possible you may have to add back --spawn_strategy=local

**Pull request merged**